File: programmers_re/스택_큐/프린터.py
def solution(priorities, location):
    answer = 0
    max_ = max(priorities)
    q = [[i,j] for i,j in enumerate(priorities)]

    while q:
        print(q)
        f,e = q.pop(0)
        if e == max_:
            if f == location:
                return answer
            answer += 1
            max_ = max(q,key=lambda x:x[1])[1]

        else:
            q.append([f,e])

    return answer

# priorities = [2, 1, 3, 2]
# priorities = [1, 1, 9, 1, 1, 1]
priorities = [1, 1, 1, 1, 1, 1, 1]

# location = 2
# location = 0
location = 6


print(solution(priorities,location))

File: programmers_java/StackQueue/프린터.java
package programmers.StackQueue;

import java.util.Collections;
import java.util.LinkedList;
import java.util.Queue;

public class 프린터 {

	public static void main(String[] args) {

//		int[] priorities = {2,1,3,2};
		int[] priorities = {1, 1, 9, 1, 1, 1};
		int location = 0;
		
		System.out.println(solution(priorities, location));
		

	}
	
    public static int solution(int[] priorities, int location) {
        int answer = 1;
        
        Queue<Integer> q = new LinkedList<Integer>();
        for (int i:priorities) {
        	q.add(i);
        }
        int max_ = Collections.max(q);
        
        while(!q.isEmpty()) {
        	System.out.println(q.toString() + " , " + location);
        	int now = q.poll();
        	if(now == max_) {
        		if(location == 0) {
        			return answer;
        		}
        		location--;
        		answer++;
        		max_ = Collections.max(q);
        	} else {
        		q.add(now);
        		location--;
        		if(location < 0) {
        			location = q.size()-1;
        		}
        	}
        }
        
        return answer;
    }
}
